currency.py

```diff
@@ -8,71 +8,72 @@ def EXIT(message, exit_code):
     sys.exit(exit_code)
 
 
-def date_check(date):    #we just need to check if date in range [1992-07-01;<today>] dmy
+def date_check(dateinp):
+    min_date = datetime(1992, 7, 1)
+    max_date = datetime.today()
 
-    lst[0] = int(lst[0])
-    lst[1] = int(lst[1])
-    lst[2] = int(lst[2])
-    today = datetime.today()
-
-    if date[0] > today.day and date[2] == today.year and date[1] == today.month:
-        return False
-
-    if (date[1] > today.month and date[2] == today.year) or (date[2] == 1992 and date[1] < 7):
-        return False
-
-    if date[2] < 1992 or date[2] > today.year:
-        return False
-
-    return True
+    if min_date <= dateinp <= max_date:
+        return True
+    return False
 
 
 EXIT_SUCCESS = 0
 CURRENCY_ERROR = 1
 DATE_ERROR = 2
 EXIT_REFERENCE = 3
 INPUT_ERROR = 4
+rFlag = False
 
 if len(sys.argv) == 1 or sys.argv[1] == "-h" or sys.argv[1] == "-a":
     print(
         "usage: currency.py [valute code] [date: <format{yyyy-mm-dd}>]", "\n",
         "usage example: eur 2020-01-01", "\n",
         "this script has data from 1992-07-01", "\n",
-        "other arguments: [-h] [-a] [-ec] [-exit_codes]", sep=""
+        "in default program show direct quotations", "-n"
+        "other arguments: [-h] [-a] [-e] [--exit-codes] [-r] [--reverse-quotation] [man]", sep=""
           )
-    EXIT("", EXIT_REFERENCE)
+    EXIT("help shown successful", EXIT_REFERENCE)
 elif sys.argv[1] == "--exit-codes" or sys.argv[1] == "-e":
     print(
         "exit code 0: success", "\n",
         "exit code 1: currency-code format error", "\n",
         "exit code 2: date error(not in data base or date in future)", "\n",
         "exit code 3: references", sep=""
     )
-    EXIT("", EXIT_REFERENCE)
-
-if len(sys.argv) != 3:
-    EXIT("Input should have format: <currency code> <date>.", INPUT_ERROR)
+    EXIT("help shown successful", EXIT_REFERENCE)
+elif sys.argv[1] == "man":
+    print(
+        "arguments help:", "\n",
+        "-h || -a : show basics for using program", "\n",
+        "-e || --exit-codes : show all exit codes with decoding it", "\n",
+        "-r || --reverse-quotation : show reverse quotations for currency you asked", sep=""
+    )
+    EXIT("help shown successful", EXIT_REFERENCE)
 
 try:
-    datetime.strptime(sys.argv[2], "%Y-%m-%d")
+    dateinp = datetime.strptime(sys.argv[2], "%Y-%m-%d")
+    if sys.argv[3] == "-r" or sys.argv[3] == "--reverse-quotations":
+        rFlag = True
 except ValueError:
     EXIT("Date format error. Date should have format: yyyy-mm-dd.", DATE_ERROR)
-else:
-    date = sys.argv[2].split("-")
-    date[0], date[2] = date[2], date[0]
+except IndexError:
```

Comment:
There is no need to catch this exception. Instead, you may define your variable like this:

```python
inverse_rate = len(sys.argv) > 2 and sys.argv[3] in ("-r", "--reverse-quotations")
```

You don't know this operator yet, but `in` is very helpful if you want to perform a search of some element in a list or string (this is called "containment test"). In my example, we're trying to determine if `sys.argv[3]` is presented in a list. This may help to get rid of multiple `or` operators.

So technically both examples are equivalent:
```python
sys.argv[3] == "-r" or sys.argv[3] == "--reverse-quotations"

# and
sys.argv[3] in ("-r", "--reverse-quotations")
```

Please notice that we don't use the name `rFlag` here. This is because it's not up to PEP-8 standards (it should rather be "r_flag" then) and because the name doesn't reflect the meaning of the variable